types/src/validator_verifier.rs

```diff
@@ -187,13 +186,7 @@ impl<PublicKey: VerifyingKey> ValidatorVerifier<PublicKey> {
                 .collect();
         // Fallback is required to identify the source of the problem if batching fails.
         if PublicKey::batch_verify_signatures(&hash, keys_and_signatures).is_err() {
```

Comment:
Ok I still am not 100% sure I understand this and I think we need to get some input from someone more familiar with what this code does. In other words: why do we even need to check aggregated sigs if the batch verify fails?

Reply:
@bmwill here is the rationale, because it was me who wrote this part of code, as it was requested from a PR reviewer when I originally implemented batch verification:

It's unlikely to happen, but if for some reason our batch verifier fails, but checking all of the signatures individually passes, then it's an alert that the batch_verification algorithm should be considered broken/inconsistent. The problem would be algorithmic implementation related or RNG related (note that batch_verifier uses an RNG (we're working on changing that to be 100% deterministic using Fiat Shamir).

Batch verification is an optimized algorithm (offering a 2.2x speedup), but it's true I have never seen inconsistencies between batching and sequential.

That said, I'm personally fine if we remove the inconsistency check. Actually, the worst thing that could happen is someone manages to pass the batch_verifier (while sequential would fail) and we cannot capture that, because optimistically we always batch_verify first.

Reply:
Thanks for the very detailed explanation :)